flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/LegacyFetcher.java

```diff
@@ -356,29 +357,24 @@ public void run() {
 				// make sure that all partitions have some offsets to start with
 				// those partitions that do not have an offset from a checkpoint need to get
 				// their start offset from ZooKeeper
-				
-				List<FetchPartition> partitionsToGetOffsetsFor = new ArrayList<>();
+				{
```

Comment:
I'm opening a new scope for the `partitionsToGetOffsetsFor` variable.
I'm using another list with the same name later in the code. WIth a new scope for the operation here, I can use the same variable name later again.